src/java.base/share/classes/com/sun/crypto/provider/ChaCha20Poly1305Parameters.java

```diff
@@ -206,8 +206,7 @@ protected byte[] engineGetEncoded(String encodingMethod)
     protected String engineToString() {
         String LINE_SEP = System.lineSeparator();
         HexDumpEncoder encoder = new HexDumpEncoder();
-        StringBuilder sb = new StringBuilder(LINE_SEP + "nonce:" +
-                LINE_SEP + "[" + encoder.encodeBuffer(nonce) + "]");
-        return sb.toString();
+        return LINE_SEP + "nonce:" +
+                LINE_SEP + "[" + encoder.encodeBuffer(nonce) + "]";
```

Comment:
This is probably not a performance sensitive method, but another fix would be to keep `StringBuilder` but call append() for each part of the String. Causes fewer objects to be created. I could accept either fix though unless there is an obvious performance issue.

Reply:
I used JMH to benchmark both fixes and they measured the same. Then I remembered that the java compiler converts + operations to StringBuilder's append. I'll keep the proposed fix.

Reply:
Yes, you are right.